update-repo.sh

```diff
@@ -106,36 +109,54 @@ done \
 
 # Commit changes and create PullRequest
 
-files="$(git ${git_common_opt} diff --name-only \
-  | sed 's/^/- /;s|/|\\/|g;s/$/\\n/g' | tr -d '\n')"
-
 git ${git_common_opt} add ros
 if git ${git_common_opt} diff --cached --exit-code; then
   echo "No update found"
 else
+  files="$(git ${git_common_opt} diff --cached --name-only)"
+  pr_body_file=$(mktemp)
+  echo "Updates found in rosdistro" > ${pr_body_file}
+
+  for file in ${files}; do
+    base_uri=$(
+      . aports/${file}
+      echo "${rosinstall}" | sed -n 's/^\s*uri: \(\S*\)$/\1/p' | sed 's/\.git//'
+    )
+    new_tag=$(
+      . aports/${file}
+      echo "${rosinstall}" | sed -n 's/^\s*version: \(\S*\)$/\1/p'
+    )
+    old_tag=$(
+      . aports.prev/${file}
+      echo "${rosinstall}" | sed -n 's/^\s*version: \(\S*\)$/\1/p'
+    )
+    diff_uri="${base_uri}/compare/${old_tag}...${new_tag}"
+
+    echo "- $(dirname ${file}) [diff](${diff_uri})" >> ${pr_body_file}
+  done
+
   date=$(date +%Y%m%d-%H%M%S)
   git ${git_common_opt} checkout -b auto-update/${ros_distro}/${ALPINE_VERSION}/${date}
   git ${git_common_opt} commit -m "${ros_distro}-${ALPINE_VERSION}: automatic update on ${date}" \
     --author="Alpine ROS aports update bot <${git_email}>"
 
   pr_user=$(dirname ${aports_slug})
-  pr_request_body=$(cat << EOF
-{
-  "title": "${ros_distro}-${ALPINE_VERSION}: automatic update on ${date}",
-  "body": "Updates found in rosdistro\\n${files}",
-  "head": "${pr_user}:auto-update\/${ros_distro}\/${ALPINE_VERSION}\/${date}",
-  "base": "master"
-}
-EOF
-)
-  echo ${pr_request_body}
+  pr_title="${ros_distro}-${ALPINE_VERSION}: automatic update on ${date}"
+  pr_head="${pr_user}:auto-update/${ros_distro}/${ALPINE_VERSION}/${date}"
+
+  echo ${pr_head}
+  echo ${pr_title}
+  cat ${pr_body_file}
   sleep 2
 
   if [ ${dry_run} == 'false' ]; then
     git ${git_common_opt} push origin auto-update/${ros_distro}/${ALPINE_VERSION}/${date}
```

Comment:
Shouldn't this be `git ${git_common_opt} push origin ${pr_head}`?

Reply:
`pr_head` contains PR source repository name. Only branch name should be passed to git command.
Added `pr_branch` variable and reuse it.